controllers/storagecluster/consumer.go

```diff
@@ -0,0 +1,189 @@
+	response, err := consumerClient.OnboardConsumer("my-ocs", getExternalStorageRequestedCapacity(instance))
+	err = r.handleOnboardConsumerErrors(instance, err)
+	if err != nil {
+		return err
+	}
+
+	if response.StorageConsumerUUID == "" || response.GrantedCapacity == "" {
+		return fmt.Errorf("response is empty")
+	}
+
+	instance.Status.ExternalStorage.ConsumerID = response.StorageConsumerUUID
+	instance.Status.ExternalStorage.GrantedCapacity = resource.MustParse(response.GrantedCapacity)
+
+	return nil
+}
+
+// offBoardConsumer makes an API call to the provider cluster for offboarding
+func offBoardConsumer(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+
+	grpcConnection, consumerClient, err := getConsumerClient(instance)
+	if err != nil {
+		return err
+	}
+	defer grpcConnection.Close()
+
+	_, err = consumerClient.OffboardConsumer(instance.Status.ExternalStorage.ConsumerID)
+	return r.handleOffboardConsumerErrors(instance, err)
+}
+
+// upscaleOrDownscaleConsumer makes an API call to the provider cluster to update the capacity
+func upscaleOrDownscaleConsumer(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+
+	grpcConnection, consumerClient, err := getConsumerClient(instance)
+	if err != nil {
+		return err
+	}
+	defer grpcConnection.Close()
+
+	response, err := consumerClient.UpdateCapacity(
+		instance.Status.ExternalStorage.ConsumerID,
+		getExternalStorageRequestedCapacity(instance))
+	err = r.handleUpdateCapacityErrors(instance, err)
+	if err != nil {
+		return err
+	}
+
+	if response.GrantedCapacity == "" {
+		return fmt.Errorf("response is empty")
+	}
+
+	instance.Status.ExternalStorage.GrantedCapacity = resource.MustParse(response.GrantedCapacity)
+
+	return nil
+}
+
+// getExternalConfigFromProvider makes an API call to the provider cluster for json blob
+// and save it in a global var (externalOCSResources) to use the same json again and again
+// It also skips the API call if json is already available
+func getExternalConfigFromProvider(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) ([]ExternalResource, error) {
+
+	if externalOCSResources[instance.UID] != nil {
+		return externalOCSResources[instance.UID], nil
+	}
+
+	grpcConnection, consumerClient, err := getConsumerClient(instance)
+	if err != nil {
+		return nil, err
+	}
+	defer grpcConnection.Close()
+
+	config, err := consumerClient.GetStorageConfig(instance.Status.ExternalStorage.ConsumerID)
+	err = r.handleGetStorageConfigErrors(instance, err)
+	if err != nil {
+		return nil, err
+	}
+
+	var externalResources []ExternalResource
+	err = json.Unmarshal(config.Data, &externalResources)
+	if err != nil {
+		return nil, err
+	}
+
+	externalOCSResources[instance.UID] = externalResources
+
+	return externalOCSResources[instance.UID], nil
+}
+
+// getConsumerClient returns the *providerClient.ConsumerClient
+func getConsumerClient(instance *ocsv1.StorageCluster) (*grpc.ClientConn, *providerClient.ConsumerClient, error) {
```

Comment:
Lets rename it to `getProviderClient`.   It was wrong in my PR as well. I've updated it.  Please rename in other places as well.

Reply:
done